src/mem/mem_interface.hh

```diff
@@ -131,7 +131,7 @@ class MemInterface : public AbstractMemory
      * ranks and banks, the burst size, and the row buffer size.
      */
     const uint32_t burstSize;
-    const uint32_t deviceSize;
+    const uint64_t deviceSize;
```

Comment:
Just a note: deviceSize should be the size of a memory chip. The whole DIMM (or MemoryController in gem5) is `deviceSize * devicesPerRank * ranksPerChannel*. 

It's possible that this is above 4 GiB, so it should be a `uint64_t`. Some devices have just 1 devices per rank and 1 ranks per channel (e.g., HBM).